### ISSUE TITLE
Let awslogs driver write to different streams depending on message source

### ISSUE THREAD

**Issue by sfussenegger** [docker/docker#17959]
An example use case would be a web server container that writes access logs to stdout and server logs to stderr.

From looking at the code, a Logger Message has a Source attribute which either is `stdout` or `stderr` (or anything else?) so it seems possible to implement.

There could be two more flags:

1. `--log-opt awslogs-stream-per-source=true`: create `${containerId}-stdout` and `${containerId}-stderr` instead of both to `${containerId}`
2. `--log-opt awslogs-stream-stderr=<log_stream_name>`: stream stderr messages to this stream

I don't know if there can be other streams, but in theory, the latter could become . `--log-opt awslogs-stream-<source_name>=<log_stream_name>` while the former creates streams for all  `${containerId}-${sourceName}`

(I had a short conversation with the author of the awslogs driver about the possibility to distinguish between messages from `stdout` and `stderr` as part of aws/amazon-ecs-agent#251)

**Comment by cpuguy83** [comment]
ping @samuelkarp 

It doesn't look like cloudwatche's API's message format supports an extra detail beyond the message and the timestamp.
Maybe support 2 event streams?

I'm not familiar with cloudwatch at all.

**Comment by sfussenegger** [comment]
@cpuguy83 that's actually what I meant. 

but on second thought, there should probably be different log groups, not only streams. Log groups can be used to stream events into Lambda functions or Kinesis streams. There isn't much sense separating the streams in the first place if both are processed equally afterwards.

so there would be an additional `--log-opt awslogs-group-stderr=<log_group_name>`

**Comment by samuelkarp** [comment]
Yes, I think separate log groups and log streams is a reasonable feature request.

**Comment by carloshgaray** [comment]
@samuelkarp any update on this? I'm working on moving my service to native AWS and it's making it impossible for me to parse my logs

**Comment by samuelkarp** [comment]
I'm not aware of anyone actively working on this.  I am happy to review changes if anyone does want to contribute this feature, however.

If you're looking for more flexible logging options that work today, you might check out [AWS FireLens](https://aws.amazon.com/blogs/aws/announcing-firelens-a-new-way-to-manage-container-logs/), which allows you to specify [filtering and routing rules](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/using_firelens.html).  You can find examples of using FireLens with Amazon ECS [here](https://github.com/aws-samples/amazon-ecs-firelens-examples).
